### DISCUSSION TITLE
Show control check in times in the region’s selected time zone

### DISCUSSION THREAD

**Discussion by rusa08** [ctnadovich/randowizard#32] [Ideas]
It would be nice if the control times could be shown in the local time zone.

**Comment by ctnadovich** [comment]
I don't understand. Control times ARE shown in the event time zone. Or should be. Have you found a bug? Please give details on how to reproduce it.

**Comment by rusa08** [comment]
When I look at rider progress, it’s shown in UTC. I’d be nice to see it in Pacific Daylight Time for us. As an additional idea, control name would be a nice to have feature in addition to control number. Sorry if I wasn’t clear. Our riders really seem to love eBrevet. LoisLois Springsteen, CFP® 831-227-6266On Mar 23, 2024, at 5:29 PM, ctnadovich ***@***.***> wrote:﻿
I don't understand. Control times ARE shown in the event time zone. Or should be. Have you found a bug? Please give details on how to reproduce it.

**Reply by ctnadovich** [reply]
How are you looking at rider progess? On the public page?

  `https://randonneuring.org/index.php/checkin_status/<event-ID> `

That's the page you get to when clicking the check-ins icon from the event_info page. 

On the checkin_status page checkins are shown in the event timezone. 

By the way, it's the EVENT timezone. Not your local timezone wherever you happen to be.  But they are definitely **not** in UTC.

Ore are you looking on the RBA-only database editing page. In the internal RBA database checkins are UTC.  The RBA only page shows UTC. I would think no RBA would use that internal database page for anything unless editing the times -- and why would somebody edit the times?  Maybe I should get rid of the internal database page so no confusion (or temptation) is created. Not sure. Or maybe put a big FOR RBA USE ONLY -- IF YOU WANT TO SEE CHECKINS NORMALLY USE THE checkin_status PAGE -- notice on it. 

There IS an issue already entered for possibly making the internal database edit page show  event timezones but this can be confusing and tricky in a different way. Best keep the database UTC.  And I'm not sure it's needed because the public pages all show event (or control) timezone.

**Comment by rusa08** [comment]
I had no idea about the functions in the “event info” page. I had not discovered those. I was using the manage-checkins function to see where everyone was. I wasn’t interested in editing anything and I don’t think I ever would be. I only wanted to see where people were. Perhaps I missed the documentation on how to follow rider progress. LoisLois Springsteen, CFP® 831-227-6266On Mar 24, 2024, at 4:09 AM, ctnadovich ***@***.***> wrote:﻿
How are you looking at rider progess? On the public page?
https://randonneuring.org/index.php/checkin_status/<event-ID> 
That's the page you get to when clicking the check-ins icon from the event_info page.
On the checkin_status page checkins are shown in the event timezone.
By the way, it's the EVENT timezone. Not your local timezone wherever you happen to be.  But they are definitely not in UTC.
Ore are you looking on the RBA-only database editing page. In the internal RBA database checkins are UTC.  The RBA only page shows UTC. I would think no RBA would use that internal database page for anything unless editing the times -- and why would somebody edit the times?  Maybe I should get rid of the internal database page so no confusion (or temptation) is created. Not sure. Or maybe put a big FOR RBA USE ONLY -- IF YOU WANT TO SEE CHECKINS NORMALLY USE THE checkin_status PAGE -- notice on it.
There IS an issue already entered for possibly making the internal database edit page show  event timezones but this can be confusing and tricky in a different way. Best keep the database UTC.  And I'm not sure it's needed because the public pages all show event (or control) timezone.

**Reply by ctnadovich** [reply]
Yes, the documentation mentions the linkable public pages. Check out item 7 in the [Quick Start Guide](https://randonneuring.org/index.php/about/quick_start).  

The idea is that you would link these pages publicly from your club website so that everyone could look at them. 

LINK: With your event published at randonneuring.org, if you so desire you can put links on your club/regional web page that direct people to the event info and roster info pages at randonneuring.org. The URLs to link are

    Past and Future Events for your region: https://randonneuring.org/regional_events/<ACP CLUB CODE>
    Future Events JSON format: https://randonneuring.org/ebrevet/future_events/<ACP CLUB CODE>
    Info about a specific event: https://randonneuring.org/event_info/<EVENT CODE>
    Rider roster and results: https://randonneuring.org/roster_info/<EVENT CODE>
    Control check ins: https://randonneuring.org/checkin_status/<EVENT CODE>

where the <EVENT CODE> is the unique identifying code for the event that combines your ACP Club Code with an event ID number (eg 905106-123). These event codes can be seen listed in the event manager and elsewhere on randonneuring.org. If you don't like the style of these pages, you can use the advanced 'style_html', 'header_html', and 'footer_html' fields in the region settings to tune the styling of the pages to match your home website.

**Reply by ctnadovich** [reply]
Nevertheless, you aren't the first person to be confounded a bit by "Manage Checkins". I'll look into clarifying the purpose of that page, or do something to make it all clearer. Thank you for the report.

**Comment by rusa08** [comment]
Thanks, Chris!Lois Springsteen, CFP® 831-227-6266On Mar 24, 2024, at 12:48 PM, ctnadovich ***@***.***> wrote:﻿
Nevertheless, you aren't the first person to be confounded a bit by "Manage Checkins". I'll look into clarifying the purpose of that page, or do something to make it all clearer. Thank you for the report.

**Comment by ctnadovich** [comment]
I made some changes on the event manager page that should better show which pages are for watching checkins versus messing with the raw data.  Hope this helps.

**Comment by rusa08** [comment]
Thanks, Chris! It looks like the elapsed time for each rider is based on when they checked in to eBrevet. Should the clock start at the advertised start time instead? While the riders have an hour to begin, their elapsed time should start when the event starts. Do I have that wrong?LoisLois Springsteen, CFP® 831-227-6266On Mar 25, 2024, at 6:47 AM, ctnadovich ***@***.***> wrote:﻿
I made some changes on the event manager page that should better show which pages are for watching checkins versus messing with the raw data.  Hope this helps.

**Comment by ctnadovich** [comment]
The elapsed time for a PRE-RIDE is based on when the riders started -- when they check in to the first control. On the other hand, for ordinary mass start events, all riders get the same start time. For a typical mass start there is a one hour grace period BEFORE the open time of the start where riders can check in -- of course they also get the usual hour AFTER the open time of the start. No matter when riders check in for a typical mass-start event, so long as it's within that plus/minus one hour window, they all get the same official start time and their elapsed time is based on this common start time.

**Comment by rusa08** [comment]
Thanks for explaining that. LoisLois Springsteen, CFP® 831-227-6266On Mar 26, 2024, at 4:41 PM, ctnadovich ***@***.***> wrote:﻿
The elapsed time for a PRE-RIDE is based on when the riders started -- when they check in to the first control. On the other hand, for ordinary mass start events, all riders get the same start time. For a typical mass start there is a one hour grace period BEFORE the open time of the start where riders can check in -- of course they also get the usual hour AFTER the open time of the start. No matter when riders check in for a typical mass-start event, so long as it's within that plus/minus one hour window, they all get the same official start time and their elapsed time is based on this common start time.
